Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -1044,12 +1049,16 @@ private bool ExecuteOnAllServicesReverseOrder(Action<IMixedRealityService> execu
                     return false;
                 }
 
-                var services = MixedRealityServiceRegistry.GetAllServices();
-                int length = services.Count;
-
-                for (int i = length - 1; i >= 0; i--)
+                try
+                {
+                    foreach (var service in MixedRealityServiceRegistry.GetAllServices().Reverse())
```

Comment:
Definitely going back to for loop in this case is better - as it stands this change adds more memory allocation on each frame.